zebra/zebra_dplane.c

```diff
@@ -1457,10 +1452,8 @@ static int dplane_ctx_ns_init(struct zebra_dplane_ctx *ctx,
 /*
  * Initialize a context block for a route update from zebra data structs.
  */
-static int dplane_ctx_route_init(struct zebra_dplane_ctx *ctx,
-				 enum dplane_op_e op,
-				 struct route_node *rn,
-				 struct route_entry *re)
+int dplane_ctx_route_init(struct zebra_dplane_ctx *ctx, enum dplane_op_e op,
```

Comment:
I'm good with exposing this, but I think we could make the path more useful by ensuring we get a memset in the path, so we can reuse a ctx in the rib walk.

Reply:
What happens with the allocated data in that function?

Example:

```c
	/* Copy nexthops; recursive info is included too */
	copy_nexthops(&(ctx->u.rinfo.zd_ng.nexthop),
		      re->nhe->nhg->nexthop, NULL);
```